internal/command/launch/launch.go

```diff
@@ -349,13 +351,69 @@ func run(ctx context.Context) (err error) {
 	if !(srcInfo == nil || srcInfo.SkipDatabase || flag.GetBool(ctx, "no-deploy") || flag.GetBool(ctx, "now")) {
 		confirmPg, err := prompt.Confirm(ctx, "Would you like to set up a Postgresql database now?")
 		if confirmPg && err == nil {
-			LaunchPostgres(ctx, appConfig.AppName, org, region)
+			db_app_name := fmt.Sprintf("%s-db", appConfig.AppName)
+			should_attach_db := false
+
+			if apps, err := client.GetApps(ctx, nil); err == nil {
+				for _, app := range apps {
+					if app.Name == db_app_name {
+						msg := fmt.Sprintf("We found an existing Postgresql database with the name %s. Would you like to attach it to your app?", app.Name)
+						confirmAttachPg, err := prompt.Confirm(ctx, msg)
+
+						if confirmAttachPg && err == nil {
+							should_attach_db = true
+
+						}
+
+					}
+
+				}
+
+			}
+
 			options["postgresql"] = true
+
+			if should_attach_db {
+				current_time := time.Now().Nanosecond()
+				db_user := fmt.Sprintf("%s-%d", db_app_name, current_time)
+				fmt.Println(db_user)
+				err = postgres.AttachCluster(ctx, postgres.AttachParams{
+					PgAppName: db_app_name,
+					AppName:   appConfig.AppName,
+					DbUser:    db_user,
+				})
+
+				if err != nil {
+					msg := `Failed attaching %s to the Postgres cluster %s: %w.\nTry attaching manually with 'fly postgres attach --app %s %s'\n`
```

Comment:
what do you think about offering to stop `fly launch` if postgres was requested and we failed to attach? for example, a yes/no prompt that asks?

I know for some frameworks they automatically make sqlite databases if they don't find database settings in the environment, so developers might like the opportunity to bail out if this step fails.

Reply:
That sounds like a good idea. Do you think we should do it for launching postgres as well?